android/javatests/org/chromium/chrome/browser/BytecodeTest.java

```diff
@@ -1691,6 +1699,14 @@ public void testConstructorsExistAndMatch() throws Exception {
                         View.OnClickListener.class,
                         OneshotSupplier.class,
                         DoubleConsumer.class));
+        Assert.assertTrue(
+                constructorsMatch(
+                        "org/chromium/chrome/browser/undo_tab_close_snackbar/UndoBarController",
+                        "org/chromium/chrome/browser/undo_tab_close_snackbar/BraveUndoBarController",
+                        Context.class,
+                        TabModelSelector.class,
+                        SnackbarManageable.class,
+                        Supplier.class));
```

Comment:
you will need proguard rules similar to that one https://github.com/brave/brave-core/blob/1.69.x/android/java/apk_for_test.flags#L20 for the tests otherwise the class is removed in the test apk in Release mode(CI will fail).